pkg/record/handler/save.go

```diff
@@ -522,10 +525,44 @@ func RecordSaveHandler(req *RecordModifyRequest, resp *RecordModifyResponse) (er
 
 type recordFunc func(*record.Record) skyerr.Error
 
-func executeRecordFunc(recordsIn []*record.Record, errMap map[record.ID]skyerr.Error, rFunc recordFunc) (recordsOut []*record.Record) {
+// executeFuncInTx ensure provided function to be run in a transaction
+// so if atomic is false, it would begin a new transaction
+func executeFuncInTx(
+	txContext db.TxContext,
+	atomic bool,
+	do func() error,
+) (err error) {
+	// Wrap function in transaction when inatomic
+	// because when atomic, there would be a transaction wrapped across all functions
+	if !atomic {
+		txErr := db.WithTx(txContext, func() error {
+			return do()
+		})
+		if txErr != nil {
+			err = skyerr.MakeError(txErr)
+		}
+
+		return
+	}
+
+	err = do()
```

Comment:
Should we also wrap this error with `skyerr.MakeError`? So the behaviour of atomic and non-atomic is the same